spec/lib/task-data/schemas/schema-ut-helper.js

```diff
@@ -4,42 +4,190 @@
+ * @param {Array<String>} jobNames - All job injectable names, it is used to check whether the
+ *  schema's describedJob points to an existing job.
+ */
+function validateSchemaDefinition(schema, jobNames) {
+    before('Check schema definition', function() {
+        expect(schema).to.be.a('object');
+        expect(schema).to.have.property('id').to.be.a('string');
+        expect(schema).to.have.property('describeJob').to.be.a('string');
+        expect(schema).to.have.property('copyright').to.be.a('string');
+        expect(schema).to.have.property('title').to.be.a('string');
+        expect(schema).to.have.property('description').to.be.a('string');
+        if (jobNames.indexOf(schema.describeJob) < 0) {
+            throw new Error("The describeJob " + schema.describeJob + " doesn't exist!");
         }
     });
+}
+
+/**
+ * A Helper Class for Schema Unit-Testing
+ * @param {String} testfile - The file path of target schema
+ * @param {Object} [canonicalData] - The canonical data for target schema
+ * @param {Boolean} [skipTaskSchemaDefValidation=false] - True to skip the validation for task
+ * schema definition itself.
+ * @param {Boolean} [skipCommonOptionsValidation=false] - True to skip the common task options
+ * validation, this depends the the canonical data is specified
+ */
+function SchemaUnitTestHelper(testfile, canonicalData, skipTaskSchemaDefValidation,
+                              skipCommonOptionsValidation) {
+    var result = init();
+    var self = this;
+    this.validator = result.validator;
+    this.schema = helper.require(testfile);
+    this.canonicalData = canonicalData;
+
+    if (!skipTaskSchemaDefValidation) {
+        validateSchemaDefinition(this.schema, result.jobNames);
+    }
+
+    if (this.canonicalData) {
+        before('Validate canonical data', function(done) {
+            self._validate(self.canonicalData, true, done);
+        });
+
+        if (!skipCommonOptionsValidation) {
+            self.validateCommonOptions();
+        }
+    }
+}
+
+/**
+ * validate data against the target schema
+ * @memberof SchemaUnitTestHelper
+ * @param {Object} data - The data to be validated.
+ * @param {Boolean} expected - true if expect data conforms to schema, otherwise expect violation.
+ * @param {Function} done - The mocha unit-test hook "done" function
+ */
+SchemaUnitTestHelper.prototype._validate = function(data, expected, done) {
+    return done(validateData(this.validator, this.schema.id, data, expected));
+};
+
+/**
+ * validate task common options
+ * @memberof SchemaUnitTestHelper
+ * @param {Object} [overrideCanonicalData] - A particular canonical data to override the default one
+ */
+SchemaUnitTestHelper.prototype.validateCommonOptions = function(overrideCanonicalData) {
+    var self = this;
+    var canonical = overrideCanonicalData || self.canonicalData;
+    describe('validate task common options', function() {
+        [36000, 0, -1].forEach(function(value) {
```

Comment:
I think it would be better to do a `forEach` loop of expect statements within a single `it` block, rather than dynamically generating 36000 `it` blocks, otherwise I believe the stdout from running the tests is going to be HUGE.